### PR TITLE
Refactor cardbot package to be discord-bot package and a more generic library

### PR DESCRIPTION
**Pull request by lukemelia**
CS-2204

### CODE REVIEW

**Comment by lukemelia** [comment]
@habdelra The bot-tests failed for me locally with the same error as seen on CI but then they started working locally and I'm not sure what changed. Any ideas?

**Comment by habdelra** [comment]
Although, this error: ```Error: Debug Failure. Output generation failed
    at Object.transpileModule (/home/runner/work/cardstack/cardstack/node_modules/typescript/lib/typescript.js:131715:29)
    at getOutput (/home/runner/work/cardstack/cardstack/node_modules/ts-node/src/index.ts:947:25)
    at Object.compile (/home/runner/work/cardstack/cardstack/node_modules/ts-node/src/index.ts:968:32)
    at Object.m._compile (/home/runner/work/cardstack/cardstack/node_modules/ts-node/src/index.ts:1056:42)
    at Module._extensions..js (internal/modules/cjs/loader.js:1114:10)```
is _facsinating_

**Comment by habdelra** [comment]
@lukemelia yeah! your tests are all passing now (with no weirdness)

File: packages/hub/main.ts [hunk 1]
```diff
@@ -302,22 +302,22 @@ export async function bootWorker() {
   await runner.promise;
 }
 
-export class HubBot {
+export class HubBotManager {
```
[the remark this answers is not in the document]
Comment (habdelra) [reply packages/hub/main.ts:305]: sounds good

**Pull request merged**